src/main/java/lotto/domain/Lotto.java

```diff
@@ -0,0 +1,50 @@
+package lotto.domain;
+
+import static lotto.domain.constant.LottoNumberConstant.LOTTO_SIZE;
+import static lotto.domain.exception.DomainExceptionCode.LOTTO_MUST_NOT_BE_DUPLICATE;
+import static lotto.domain.exception.DomainExceptionCode.LOTTO_SIZE_MUST_BE_SAME;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+
+public class Lotto {
+
+    private static final int COMBINE_MAX_SIZE = 12;
+    private final List<Integer> numbers;
+
+    public Lotto(List<Integer> numbers) {
+        validate(numbers);
+        this.numbers = numbers;
+    }
+
+    private void validate(List<Integer> numbers) {
+        LOTTO_SIZE_MUST_BE_SAME.dynamicInvokeBy(() -> LOTTO_SIZE.isNotEqualTo(numbers.size()));
+        LOTTO_MUST_NOT_BE_DUPLICATE.dynamicInvokeBy(() -> LOTTO_SIZE.isNotEqualTo(new HashSet<>(numbers).size()));
+    }
+
+    // TODO: 추가 기능 구현
+    public List<Integer> getNumbers() {
+        return Collections.unmodifiableList(this.numbers);
```

Comment:
[이 부분은 이렇게 바꿀 수 있을 것 같아요!]

```suggestion
        return numbers.stream().toList();
```

Reply:
원본과의 참조를 끊고 아예 새로운 리스트 반환하는 방식이네요!

저는 여태 `new ArrayList<>(numbers)`와 같은 방식을 사용헀는데 새로운 방법을 알게되었습니다...!

Reply:
@h-beeen 
두 코드는 조금 다르게 봐야합니다!

먼저 stream()은 내부적으로 Spliterator로 변환해주는 작업을 거치는데 이는 병렬 처리 혹은 빌딩 블록을 만들기 위해 필요한 과정으로 알고 있어요! 
그래서 별다른 연산 없이 불변 컬렉션을 제공하기 위함이라면 Collections.unmodifiableList을 사용하는 것 이 좋습니다 :) 
추가로 저는 모든 객체를 불변으로 유지하고 있기 때문에 상태가 변경되는 사이드 이펙트가 발생하지 않습니다!
다만 컬렉션을 추가/삭제하는 것은 막을 수 없기 때문에 Collections.unmodifiableList를 사용했습니다!